### ISSUE TITLE
Do we use `GridObject.rows` or `GridObject.columns` anywhere?

### ISSUE THREAD

**Issue by wkearn** [TopoToolbox/pytopotoolbox#90]
It seems like most places use `grid.shape` to get the dimensions of the array to supply to libtopotoolbox. Is there any thing that depends on the separate `rows` and `columns` attributes? If not, it might be worth removing them just to prevent confusion.

**Comment by bgailleton** [comment]
Hi @wkearn, I am not sure I use them or need them but I like being able to access to a variable through different names (e.g. if you need to match another API). A safe way to do that could be to move them as properties linked to shape?

It would mean any access to rows/columns are directly accessing/modifying shape (with the option to make them  unmodifiable too).

Example (in GridObj):
```python
...
@property
def rows(self):
    return self.shape[0]

#if you want to make it modifiable
@rows.setter
def rows(self,value):
    #Add any check you need here
    self.shape[0] = value

```

in practice it is fully hidden to users:

```python
print(grid.rows)
grid.rows = 500
```

You could even make `shape` as a property exposing a hidden self._shape that would manage all the getter/setter with extra checks if required

**Comment by wkearn** [comment]
I like this idea. I will try it out and see how it works.

**Comment by bgailleton** [comment]
It's quite common in `python` and very powerful. I use it a lot for example to standardise the change of one parameter and make it backend agnostic. Or typically if a parameter requires the re-compilation of a GPU kernel.

**Comment by wkearn** [comment]
Resolved by #93
